core/modules/catalog/store/category/actions.ts

```diff
@@ -70,7 +71,7 @@ const actions: ActionTree<CategoryState, RootState> = {
         for (let category of resp.items) {
           if (category.url_path && updateState) {
             rootStore.dispatch('url/registerMapping', {
-              url: category.url_path,
+              url: localizedDispatcherRoute('/' + category.url_path, currentStoreView().storeCode),
```

Comment:
maybe localizedDispatcherRoute could detect slash at the beginning by itself? It reduces complexity.

And also storeCode as default

Reply:
OK